File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.py
class Solution:
    def doesValidArrayExist(self, derived: List[int]) -> bool:
        n = len(derived)
        original = [0]*n
        for i in range(1,n):
            original[i] = derived[i-1]^original[i-1]
        
        for i in range(0,n):
            index = (i+1)%n
            if derived[i] != (original[i]^original[index]):
                return False
        
        return True

File: 2792-neighboring-bitwise-xor/2792-neighboring-bitwise-xor.java
class Solution {
    public boolean doesValidArrayExist(int[] derived) {
        int n = derived.length;
        int[] original = new int[n];
        for(int i=1;i<n;i++){
            original[i] = derived[i-1]^original[i-1];
        }

        for(int i=0;i<n;i++){
            int index = (i+1)%n;
            if(derived[i] != (original[i]^original[index])){
                return false;
            }
        }
        return true;
    }
}